apps/cowswap-frontend/src/legacy/hooks/useWrapCallback.ts

```diff
@@ -104,24 +105,47 @@ function getWrapDescription(
   }
 }
 
-async function wrapContractCall(wethContract: Contract, amountHex: string): Promise<TransactionResponse> {
+async function wrapContractCall(
+  wethContract: Contract,
+  amountHex: string,
+  chainId: SupportedChainId,
+): Promise<TransactionResponse> {
   const estimatedGas = await wethContract.estimateGas.deposit({ value: amountHex }).catch(_handleGasEstimateError)
   const gasLimit = calculateGasMargin(estimatedGas)
 
-  return wethContract.deposit({ value: amountHex, gasLimit })
+  const network = await wethContract.provider.getNetwork()
```

Comment:
@alfetopito @shoom3301 do you think getNetwork in the provider does the RPC called or does it return the cached data? if it returns the cached data we might still be affected by the bug

Reply:
It comes from https://github.com/ethers-io/ethers.js/blob/v5.7/packages/providers/src.ts/base-provider.ts#L1154
This method has a mechanism with is supposed to check the cached value with a fresh one from RPC using `detectNetwork` method:
https://github.com/ethers-io/ethers.js/blob/v5.7/packages/providers/src.ts/json-rpc-provider.ts#L448

This method is supposed to reset cache at the next event loop.
Buuuuuuut, we got rid of this caching in order to save RPC calls:
https://github.com/cowprotocol/cowswap/blob/develop/patches/%40ethersproject%2Bproviders%2B5.7.2.patch#L12

**In conclusion, the value is cached.**

Reply:
If we want to get a not-cached chainId, we should do:
```
await provider.send('eth_chainId')
```

Reply:
https://github.com/cowprotocol/cowswap/pull/5354